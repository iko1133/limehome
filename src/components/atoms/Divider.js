import React from "react";
import styled from "styled-components/native";

const Container = styled.View`
  background-color: darkgray;
  height: 1px;
  margin: 8px 0;
  width: 100%;
`;

const Divider = (props) => {
  return <Container />;
};

export default Divider;
