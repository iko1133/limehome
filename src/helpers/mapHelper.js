// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
