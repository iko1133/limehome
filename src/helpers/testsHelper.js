import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
