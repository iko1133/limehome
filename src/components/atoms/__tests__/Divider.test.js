import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import Divider from "../Divider";

describe("Testing Divider", () => {
  it("Renders Divider", () =>
    expect(
      renderer.create(WrapInThemeProvider(<Divider />))
    ).toMatchSnapshot());
});
