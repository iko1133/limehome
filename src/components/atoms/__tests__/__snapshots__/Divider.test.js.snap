// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Divider Renders Divider 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "darkgray",
        "height": 1,
        "marginBottom": 8,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 8,
        "width": "100%",
      },
    ]
  }
/>
`;
